product_harmonized_system_country/models/product_product.py

```diff
@@ -0,0 +1,20 @@
+# Copyright 2024 Manuel Regidor <manuel.regidor@sygel.es>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import models
+
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    def get_hs_local_code_by_country(self, country=False):
+        self.ensure_one()
+        hs_code = self.get_hs_code_recursively()
+        code = ""
+        if hs_code and country:
+            country_hs_code = hs_code.hs_code_country_ids.filtered(
+                lambda a: country.id in a.country_ids.ids
+            )
+            if country_hs_code:
```

Comment:
se tendría que comprobar que solo haya cogido un codigo?

Reply:
There's a restrictions that does not allow to have the same country in multiple country lines, but I've added the check anyway.